x-pack/plugins/apm/server/routes/settings/agent_configuration/route.ts

```diff
@@ -208,21 +287,34 @@ const agentConfigurationSearchRoute = createApmServerRoute({
       >
     | null
   > => {
-    const { params, logger } = resources;
+    const { params, logger, context, config, request } = resources;
 
     const {
       service,
       etag,
       mark_as_applied_by_agent: markAsAppliedByAgent,
     } = params.body;
 
-    const setup = await setupRequest(resources);
-    const config = await searchConfigurations({
+    const savedObjectsClient = (await context.core).savedObjects.client;
+
+    const [internalESClient, indices] = await Promise.all([
+      createInternalESClient({
+        context,
+        request,
+        debug: params.query._inspect,
+      }),
+      getApmIndices({
+        savedObjectsClient,
+        config,
+      }),
```

Comment:
Everywhere the `internalEsClient` is used we also call `getApmIndices`. They seem to belong together.